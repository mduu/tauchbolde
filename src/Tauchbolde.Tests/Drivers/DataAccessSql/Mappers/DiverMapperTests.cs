using System;
using System.Runtime.CompilerServices;
using ApprovalTests;
using ApprovalTests.Reporters;
using FluentAssertions;
using Newtonsoft.Json;
using Tauchbolde.Driver.DataAccessSql.DataEntities;
using Tauchbolde.Driver.DataAccessSql.Mappers;
using Xunit;

namespace Tauchbolde.Tests.Drivers.DataAccessSql.Mappers
{
    [UseReporter(typeof(DiffReporter))]
    public class DiverMapperTests
    {
        private readonly Guid diverId = new Guid("C35F8D49-057B-4A18-85D9-1AB3E866D3EC");

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Map()
        {
            var dataObject = new DiverData
            {
                Id = diverId,
                Education = "The Education",
                Experience = "The Experience",
                Slogan = "The Slogan",
                Fullname = "Fullname",
                Firstname = "Firstname",
                Lastname = "Lastname",
                UserId = "joe@doe.com",
                AvatarId = "TheAvatarId1234",
                FacebookId = "FacebookId1234",
                TwitterHandle = "TwitterHandle1234",
                SkypeId = "SkypeId1234",
                MemberSince = new DateTime(2017, 7, 1),
                MemberUntil = new DateTime(2019, 8, 13),
                MobilePhone = "4242 42 42",
                WebsiteUrl = "https://tauchbolde.ch",
                SendOwnNoticiations = true,
                LastNotificationCheckAt = new DateTime(2019, 8, 8, 13, 55, 00),
                NotificationIntervalInHours = 4,
            };

            var domainObject = dataObject.MapTo();
            var newDataObject = domainObject.MapTo();

            newDataObject.Should().BeEquivalentTo(dataObject);
            Approvals.VerifyJson(JsonConvert.SerializeObject(
                new
                {
                    dataObject,
                    domainObject,
                    newDataObject
                }));
        }
    }
}