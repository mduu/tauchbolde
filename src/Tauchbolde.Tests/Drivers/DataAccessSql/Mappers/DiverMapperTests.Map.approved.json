{
    "dataObject": {
        "Fullname": "Fullname",
        "Firstname": "Firstname",
        "Lastname": "Lastname",
        "MemberSince": "2017-07-01T00:00:00",
        "MemberUntil": "2019-08-13T00:00:00",
        "AvatarId": "TheAvatarId1234",
        "WebsiteUrl": "https://tauchbolde.ch",
        "TwitterHandle": "TwitterHandle1234",
        "FacebookId": "FacebookId1234",
        "SkypeId": "SkypeId1234",
        "Slogan": "The Slogan",
        "Education": "The Education",
        "Experience": "The Experience",
        "MobilePhone": "4242 42 42",
        "NotificationIntervalInHours": 4,
        "SendOwnNoticiations": true,
        "LastNotificationCheckAt": "2019-08-08T13:55:00",
        "UserId": "joe@doe.com",
        "User": null,
        "Notificationses": null,
        "Events": null,
        "Comments": null,
        "OriginalAuthorOfLogbookEntries": null,
        "EditorAuthorOfLogbookEntries": null,
        "Id": "c35f8d49-057b-4a18-85d9-1ab3e866d3ec"
    },
    "domainObject": {
        "Fullname": "Fullname",
        "Firstname": "Firstname",
        "Lastname": "Lastname",
        "MemberSince": "2017-07-01T00:00:00",
        "MemberUntil": "2019-08-13T00:00:00",
        "AvatarId": "TheAvatarId1234",
        "WebsiteUrl": "https://tauchbolde.ch",
        "TwitterHandle": "TwitterHandle1234",
        "FacebookId": "FacebookId1234",
        "SkypeId": "SkypeId1234",
        "Slogan": "The Slogan",
        "Education": "The Education",
        "Experience": "The Experience",
        "MobilePhone": "4242 42 42",
        "NotificationIntervalInHours": 4,
        "SendOwnNoticiations": true,
        "LastNotificationCheckAt": "2019-08-08T13:55:00",
        "UserId": "joe@doe.com",
        "User": null,
        "Realname": "Fullname",
        "Id": "c35f8d49-057b-4a18-85d9-1ab3e866d3ec"
    },
    "newDataObject": {
        "Fullname": "Fullname",
        "Firstname": "Firstname",
        "Lastname": "Lastname",
        "MemberSince": "2017-07-01T00:00:00",
        "MemberUntil": "2019-08-13T00:00:00",
        "AvatarId": "TheAvatarId1234",
        "WebsiteUrl": "https://tauchbolde.ch",
        "TwitterHandle": "TwitterHandle1234",
        "FacebookId": "FacebookId1234",
        "SkypeId": "SkypeId1234",
        "Slogan": "The Slogan",
        "Education": "The Education",
        "Experience": "The Experience",
        "MobilePhone": "4242 42 42",
        "NotificationIntervalInHours": 4,
        "SendOwnNoticiations": true,
        "LastNotificationCheckAt": "2019-08-08T13:55:00",
        "UserId": "joe@doe.com",
        "User": null,
        "Notificationses": null,
        "Events": null,
        "Comments": null,
        "OriginalAuthorOfLogbookEntries": null,
        "EditorAuthorOfLogbookEntries": null,
        "Id": "c35f8d49-057b-4a18-85d9-1ab3e866d3ec"
    }
}